common/txmgr/txmgr.go

```diff
@@ -675,3 +675,42 @@ func (n *NullTxManager[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) Fin
 func (n *NullTxManager[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) CountTransactionsByState(ctx context.Context, state txmgrtypes.TxState) (count uint32, err error) {
 	return count, errors.New(n.ErrMsg)
 }
+
+func (b *Txm[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) createTxnAndPruneQueue(
+	ctx context.Context,
+	txRequest txmgrtypes.TxRequest[ADDR, TX_HASH],
+	chainID CHAIN_ID,
+) (
+	tx txmgrtypes.Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE],
+	err error,
+) {
+	pruned, err := txRequest.Strategy.PruneQueue(ctx, b.txStore)
+	if err != nil {
+		return tx, err
+	}
+
+	tx, err = b.txStore.CreateTransaction(ctx, txRequest, chainID)
+	if err != nil {
+		if len(pruned) > 0 {
+			b.logger.Warnw(fmt.Sprintf("Dropped %d old transactions from transaction queue", len(pruned)),
+				"fromAddress", txRequest.FromAddress,
+				"toAddress", txRequest.ToAddress,
+				"meta", txRequest.Meta,
+				"subject", txRequest.Strategy.Subject(),
+			)
+		}
+		return tx, err
+	}
+
+	if len(pruned) > 0 {
+		b.logger.Warnw("Dropped transaction replaced by new transaction",
```

Comment:
This log in current form doesn't make too much sense.
Lets just log about the pruning, immediately after the line:
"pruned, err := txRequest.Strategy.PruneQueue(ctx, b.txStore)"

So we log that certain Txs were pruned, to make space for new incoming ones.
And then whether new Tx got added or not, can be separately dealt with.